patches/net/minecraft/network/protocol/handshake/ClientIntentionPacket.java.patch

```diff
@@ -37,13 +37,17 @@
          p_134737_.writeShort(this.port);
          p_134737_.writeVarInt(this.intention.id());
      }
-@@ -34,5 +_,9 @@
+@@ -34,5 +_,13 @@
      @Override
      public ConnectionProtocol nextProtocol() {
          return this.intention.protocol();
 +    }
 +
 +    public String getFMLVersion() {
 +        return this.fmlVersion;
++    }
++    
++    private static String getFMLVersion(final String ip) {
++        return ip.contains("\0") ? java.util.Objects.equals(ip.split("\0")[1], net.neoforged.neoforge.internal.versions.neoforge.NeoForgeVersion.getSpec()) ? net.neoforged.neoforge.internal.versions.neoforge.NeoForgeVersion.getSpec() : ip.split("\0")[1] : "NONE";
```

Comment:
I feel like we had a discussion about this at some point... and there being some doubt as to whether it potentially uses object equality checks, but given I don't fully remember I am going to bring it up again...

if `ip.split("\0")[1]` is equal to `NeoForgeVersion.getSpec()` then two bits of the ternary are equal and this entire return statement could be simplified to `return ip.contains("\0") ? ip.split("\0")[1] : "NONE";`

Reply:
After @XFactHD checked: This can be removed entirely